pkg/controllers/resources/endpoints/syncer.go

```
@@ -41,79 +36,23 @@ func (s *endpointsSyncer) Sync(ctx *synccontext.SyncContext, pObj client.Object,
 
 func (s *endpointsSyncer) ReconcileStart(ctx *synccontext.SyncContext, req ctrl.Request) (bool, error) {
-	// don't do anything for the kubernetes service
-	if req.Name == "kubernetes" && req.Namespace == "default" {
-		return true, SyncKubernetesServiceEndpoints(ctx.Context, ctx.VirtualClient, ctx.CurrentNamespaceClient, ctx.CurrentNamespace, s.serviceName)
-	}
-
-	if s.syncServiceSelector {
-		return true, nil
-	}
-	return false, nil
-}
-
-func (s *endpointsSyncer) ReconcileEnd() {}
-
-func SyncKubernetesServiceEndpoints(ctx context.Context, virtualClient client.Client, localClient client.Client, serviceNamespace, serviceName string) error {
-	// get physical service endpoints
-	pObj := &corev1.Endpoints{}
-	err := localClient.Get(ctx, types.NamespacedName{
-		Namespace: serviceNamespace,
-		Name:      serviceName,
-	}, pObj)
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			return nil
-		}
-
-		return err
-	}
+	svc := &corev1.Service{}
+	err := ctx.VirtualClient.Get(ctx.Context, types.NamespacedName{
+		Namespace: req.Namespace,
+		Name:      req.Name,
+	}, svc)
 
-	// get virtual service endpoints
-	vObj := &corev1.Endpoints{}
-	err = virtualClient.Get(ctx, types.NamespacedName{
-		Namespace: "default",
-		Name:      "kubernetes",
-	}, vObj)
 	if err != nil {
 		if kerrors.IsNotFound(err) {
-			return nil
-		}
-
-		return err
-	}
-
-	// build new subsets
-	newSubsets := pObj.DeepCopy().Subsets
-	for i := range newSubsets {
-		for j := range newSubsets[i].Ports {
-			newSubsets[i].Ports[j].Name = "https"
-		}
-		for j := range pObj.Subsets[i].Addresses {
-			newSubsets[i].Addresses[j].Hostname = ""
-			newSubsets[i].Addresses[j].NodeName = nil
-			newSubsets[i].Addresses[j].TargetRef = nil
-		}
-		for j := range pObj.Subsets[i].NotReadyAddresses {
-			newSubsets[i].NotReadyAddresses[j].Hostname = ""
-			newSubsets[i].NotReadyAddresses[j].NodeName = nil
-			newSubsets[i].NotReadyAddresses[j].TargetRef = nil
+			return true, nil
 		}
```

I think we should return an error if it's of a different type than IsNotFound.